setup/migrations/vexim_2.2_to_2.3_mysql.sql

```diff
@@ -0,0 +1,136 @@
+--
+-- MySQL script to upgrade Vexim database schema from Vexim 2.2RC1, 2.2 and 2.2.1 to Vexim 2.3
+--
+-- Uncomment the section at the bottom of this file if you used Arne Schirmacher's patch for Vexim 2.2RC1
+-- from https://www.schirmacher.de/display/INFO/improved+Vexim+frontend+and+bug+fixes
+--
+-- The following line defines a new salt prefix for password hashes.
+-- Before dropping the `clear` field, we will re-hash passwords for all users whose cleartext and encrypted
+-- passwords match using a new, more secure hashing scheme (SHA-512 by default).
+-- Below you may change this prefix if you prefer a different hashing scheme, e.g. '$2a$10$' for bcrypt
+-- (*BSD-only Blowfish hashing scheme).
+SET @NEW_PW_PREFIX='$6$';
+-- Comment out this definition if you do not want your password hashes updated. Be advised that it will
+-- leave these passwords less secure for no good reason.
+-- In case you want to store passwords in unencrypted form (PLEASE DO NOT DO THIS), uncomment the following
+-- UPDATE statement that follows instead. Note this will cut very long passwords to 48 characters. If this
+-- is a problem for you, move the uncommented statement just above the statement that re-hashes passwords.
+-- UPDATE `users` SET `crypt` = `clear`;
```

Comment:
Yes, we can shorten this. `sha512` should be doable by everybody with a current OS and we should not promote clear-text passwords at all. Perhaps we should leave a short note for people who manually deactivated the clear-text option, but that could go into a release-note or changelog as well.